File: Problems/Phase 1 - Foundations/Day 2 - Running Sum/Python3/SmiddleySolution.py
# Given an array, return a new array where each element at index i is the sum of nums[0..i].

def sum_array(arr: list[int]) -> list[int]:
    if len(arr) == 0:
        return []
    
    ans = []
    total = 0

    for value in arr:
        total = total + value
        ans.append(total)
    
    return ans

if __name__ == "__main__":
    arr = [1,5,7,3,-7,8,3,4]
    print(sum_array(arr))

File: Problems/Phase 1 - Foundations/Day 2 - Running Sum/Go/SmiddleySolution.go
// Given an array, return a new array where each element at index i is the sum of nums[0..i].

package main

import (
	"errors"
	"fmt"
)

func sumSlice(arr []int) ([]int, error) {
	if arr == nil {
		return nil, errors.New("Slice is nil")
	}
	if len(arr) == 0 {
		return []int{}, nil
	}

	ans := make([]int, 0, len(arr))
	total := 0
	for _, value := range arr {
		total = total + value
		ans = append(ans, total)
	}
	return ans, nil
}

func main() {
	arr := []int{1, 5, 7, 3, -7, 8, 3, 4}
	fmt.Println(sumSlice(arr))
}
